### DISCUSSION TITLE
Reason behind application_name = cnpg_metrics_collector in metrics collector

### DISCUSSION THREAD

**Discussion by RickVenema** [cloudnative-pg/cloudnative-pg#2929] [Q&A]
I was wondering what the reasoning behind the application_name = cnpg_metrics_collector in the createMonitoringTx function in pkg/management/postgres/metrics/collector.go is

I tried creating a custom metric for TLS monitoring, however I could only get cnpg_metrics_collector psql connection and not the replication connections for example.

Got it working with setting it to the default value in my configmap with my query, however its a dirty sulotion. And it made me wonder why this application_name was hardcoded set to cnpg_metrics_collector

**Comment by gbartolini** [comment]
Hi @RickVenema, thanks for reaching out. Could you please help me understand the issue here, and most importantly the goal? Apologies.

**Reply by RickVenema** [reply]
When I add a custom metric that uses application_name in the metric, the only results that show up are the ones with application_name = cnpg_metrics_collector. I used a custom function to set application_name on default and then the desired metrics show up. (I wanted to have a metric for all the connections that have TLS enabled on them)

This seems like a very dirty fix with a custom function, and I wondered what the reason for a hardcoded application_name = cnpg_metrics_collector was. 

The only difference when having it on default looks like that some replication metrics show more information.

I dont really have a goal other than to understand the operator more in depth. The custom metrics for TLS were to try working with adding my own metrics.
